src/mcp_panther/panther_mcp_core/client.py

```diff
@@ -296,7 +299,15 @@ def _build_url(self, path: str) -> str:
 
         Returns:
             str: The complete URL with base URL and path
+
+        Raises:
+            ValueError: If the base URL is not set
         """
+        if not self._base_url:
+            raise ValueError(
+                "Panther REST API base URL not set. Please check your PANTHER_INSTANCE_URL environment variable or ensure the Panther instance is accessible."
+            )
```

Comment:
hmm this feels too specific for me. I think we should handle this better closer to where we are using this. 

What likely happened is when I introduced the 'enhanced' resolution logic, it should've better handled this failure. 

I don't think we should be referencing rest anything here.

If you want we can pull this out and address it in a dedicated PR.

Reply:
Sounds great, I'll do that.